src/video/x11/SDL_x11window.c

```diff
@@ -1888,6 +1871,39 @@ X11_FlashWindow(_THIS, SDL_Window * window, SDL_FlashOperation operation)
     return 0;
 }
 
+bool SDL_X11_SetWindowTitle(Display* display, Window xwindow, char* title) {
+    Atom _NET_WM_NAME = X11_XInternAtom(display, "_NET_WM_NAME", False);
+    XTextProperty titleprop;
+    int conv = X11_XmbTextListToTextProperty(display, (char**) &title, 1, XStdICCTextStyle, &titleprop);
+    Status status;
+
+    if (conv == 0) {
+        X11_XSetTextProperty(display, xwindow, &titleprop, XA_WM_NAME);
+        X11_XFree(titleprop.value);
+    // we know this can't be a locale error as we checked X locale validity in X11_VideoInit
+    } else if (conv <= 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_VIDEO, "X11 reporting it's out of memory");
+        return EXIT_FAILURE;
+    } else { // conv >= 0
+        SDL_LogWarn(SDL_LOG_CATEGORY_VIDEO, "%d characters were not convertable to the current locale!", conv);
```

Comment:
This function should return int and do this instead:
return SDL_SetError("%d characters were not convertable to the current locale!", conv);

Reply:
This is the only review comment I'm leery about and wish to discuss further. Because this is a fallback representation, I think we should be just a tad more fault tolerant regarding bad stuff in it. X11 will always put a string there, it's just, it'll report to you how many characters it _failed_ to encode. Note this:

> If the supplied text is not fully convertible to the specified encoding, the functions return the number of unconvertible characters. Each unconvertible character is converted to an implementation-defined and encoding-specific default string. 

X11 will put the `XDefaultString()` value anywhere that it couldn't do a requested decode. Consider, for example, a user on a ja_JP.SHIFT-JIS encoding Xorg (yes, in 2021, humor me for the sake of an example :laughing:). If they go on an Arabic webpage in an SDL2 web browser (yes my example just invented a competitor to Chrome, why not, it's causing this whole PR in the first place hehe), that'd be a failure to encode some characters, but shouldn't error out the whole title I don't think. I think it's good information to have if it starts going wrong on us, but there are legit reasons (in the early 2000's at least…) it might happen.

So I'd ask you to reconsider this request.